src/webgl/3d_primitives.js

```diff
@@ -1532,4 +1532,50 @@ p5.RendererGL.prototype.curveVertex = function() {
   }
 };
 
+p5.RendererGL.prototype.image = function(
+  img,
+  sx,
+  sy,
+  sWidth,
+  sHeight,
+  dx,
+  dy,
+  dWidth,
+  dHeight
+) {
+  this._pInst.push();
+
+  this._pInst.texture(img);
+  this._pInst.textureMode(constants.NORMAL);
+
+  var u0 = 0;
+  if (sx <= img.width) {
+    u0 = sx / img.width;
+  }
+
+  var u1 = 1;
+  if (sx + sWidth <= img.width) {
+    u1 = (sx + sWidth) / img.width;
+  }
+
+  var v0 = 0;
+  if (sy <= img.height) {
+    v0 = sy / img.height;
+  }
+
+  var v1 = 1;
+  if (sy + sHeight <= img.height) {
+    v1 = (sy + sHeight) / img.height;
+  }
+
+  this.beginShape();
+  this.vertex(dx, dy, 0, u0, v0);
+  this.vertex(dx + dWidth, dy, 0, u1, v0);
+  this.vertex(dx + dWidth, dy + dHeight, 0, u1, v1);
+  this.vertex(dx, dy + dHeight, 0, u0, v1);
+  this.endShape(constants.CLOSE);
+
+  this._pInst.pop();
+};
+
```

Comment:
I think that unit tests should be added for this new method. In can be tricky to catch lack of unit test coverage for new functions down the road. If you don't have the bandwidth for this then an issue should be opened to track that these unit tests don't exist yet. So one of two options:
- add a new simple unit test suite to the [RendererGL](https://github.com/processing/p5.js/blob/master/test/unit/webgl/p5.RendererGL.js) tests. Inside you can have a simple test that [preloads an existing small image asset like this](https://github.com/processing/p5.js/blob/master/test/unit/webgl/p5.Texture.js#L13). Then you should call your new image function with `myp5.image(args)` and use get pixel to make sure that the image is applied as expected. You can test the result of (get(x,y)) in a separate sketch so you know what to assert is the return value of that function when the image is applied properly. If you decide to go this route and need more help feel free to let me know.
- Alternatively, you can open a new issue explaining what needs to happen with adding the unit test and someone else can take care of this. Either option is fine just don't want to lose track of this as a to-do.